D-lama-service/d-lama-service/Controllers/DataPointController.cs

```diff
@@ -606,16 +609,30 @@ public async Task<IActionResult> RemoveLabelDataPoint(int projectId, int dataPoi
         [HttpGet("{projectId:int}/GetLabeledData")]
         public async Task<IActionResult> GetLabeledDataForProject(int projectId) 
         {
-            var project = await GetProjectWithOwnerCheckAsync(projectId, e => e.DataPoints);
-            var labeledDataModelList = new Dictionary<int,LabeledDataPointModel>();  
+            var project = await GetProjectWithOwnerCheckAsync(projectId, e => e.DataPoints, e => e.Labels);
+            var labeledDataModelDic = new Dictionary<int,LabeledDataPointModel>();  
+            var userIds = new HashSet<int>();
             foreach (var dataPoint in project.DataPoints) 
             {
                 var dataPointStats = await _unitOfWork.LabeledDataPointRepository.GetLabeledDataPointStatisticAsync(dataPoint.Id);
-                labeledDataModelList.Add(dataPoint.Id, new LabeledDataPointModel(dataPointStats));
+                labeledDataModelDic.Add(dataPoint.DataPointIndex, new LabeledDataPointModel(dataPointStats));
             }
-            return Ok(labeledDataModelList);
+            var projectModel = new ProjectModel(project);
+
+            var userCompletion = await _unitOfWork.LabeledDataPointRepository.GetLabeledDataPointsProcessOfUser(project.DataPoints.Select(e => e.Id));
+            var labellers = new List<LabellerModel>();
+            foreach(var user in userCompletion)
+            {
+                var userInfo = (await _unitOfWork.UserRepository.GetAsync(user.Key));
+                labellers.Add(new LabellerModel { Id = userInfo!.Id, FirstName = userInfo.FirstName, LastName = userInfo.LastName, Percentage = user.Value });
+            }
+
+            var viewModel = new LabeledDataModel { DataPoints = labeledDataModelDic, Project = projectModel, Labellers = labellers };
+            return Ok(viewModel);
         }
 
+
+
```

Comment:
Also a clean code issue. I will mark it here in case I'll find a bigger issue in this review and you have to fix the code before merging into main, but if I only find clean code issues, we do not have to resolve them now.